fl4health/clients/ensemble_client.py

```diff
@@ -82,13 +84,16 @@ def train_step(self, input: torch.Tensor, target: torch.Tensor) -> Tuple[Trainin
         that we have to do backward passes on and multiple optimizers to update parameters each train step.
 
         Args:
-            input (torch.Tensor): The input to be fed into the model.
+            input (TorchInputType): The input to be fed into the model.
+            TorchInputType is simply an alias for the union of torch.Tensor and
+            Dict[str, torch.Tensor].
             target (torch.Tensor): The target corresponding to the input.
 
         Returns:
             Tuple[TrainingLosses, Dict[str, torch.Tensor]]: The losses object from the train step along with
                 a dictionary of any predictions produced by the model.
         """
+        assert isinstance(input, torch.Tensor)
```

Comment:
Any reason that we have to do this asset here? The basic_client predict method should be able to handle a `Dict[str, torch.Tensor]` I think?

Reply:
I only did this assert because the `forward()` method of `EnsembleModel` only takes in `torch.Tensor` as its input, and the model of `ensemble_client` has type `EnsembleModel`. For the same reason I also added an assert for Fenda and Moon. Do you think it would be better to do `self.model(**input)` instead?

Reply:
Ah I see. No, let's leave that for now. We can refactor when necessary.